packages/router/src/navigation_canceling_error.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {NavigationCancellationCode} from './events';
+import {NavigationBehaviorOptions} from './models';
+import {isUrlTree, UrlSerializer, UrlTree} from './url_tree';
+
+export const NAVIGATION_CANCELING_ERROR = 'ngNavigationCancelingError';
+
+export type NavigationCancelingError =
+    Error&{[NAVIGATION_CANCELING_ERROR]: true, cancellationCode: NavigationCancellationCode};
+export type RedirectingNavigationCancelingError = NavigationCancelingError&{
+  url: UrlTree,
+  navigationBehaviorOptions?: NavigationBehaviorOptions,
+                           cancellationCode: NavigationCancellationCode.Redirect
```

Comment:
The formatting looks a bit unusual. May be try adding a comma at the end (that might trigger a reformat)?
```suggestion
  navigationBehaviorOptions?: NavigationBehaviorOptions,
                           cancellationCode: NavigationCancellationCode.Redirect,
```